unix/imageUtil_unix.ml

```diff
@@ -0,0 +1,95 @@
+open ImageUtil
+
+(** [chop_extension' fname] is the same as [Filename.chop_extension fname] but
+    if [fname] does not have an extension, [fname] is returned instead of
+    raising [Invalid_argument]. *)
+let chop_extension' fname =
+  try Filename.chop_extension fname
+  with _ -> fname
+
+
+(** [get_extension fname] returns the extension of the file [fname]. If the
+    file does not have an extension, [Invalid_argument] is raised. *)
+let get_extension fname =
+  let baselen = String.length (chop_extension' fname) in
+  let extlen  = String.length fname - baselen - 1 in
+  if extlen <= 0
+  then let err = Printf.sprintf "No extension in filename '%s'." fname in
+       raise (Invalid_argument err)
+  else String.sub fname (baselen + 1) extlen
+
+
+(** [get_extension' fname] is the same as [get_extension fname] but if [fname]
+    does not have an extension, the empty string is returned and no exception
+    is raised. *)
+let get_extension' fname =
+  try get_extension fname
+  with _ -> ""
+
+
+(*
+ * Reads all the lines in the channel by calling input_line.
+ * Returns a list of strings.
+ *)
+let lines_from_channel ich =
+  let lines = ref [] in
+
+  let rec intfun () =
+    try
+      let l = input_line ich in
+      lines := l :: !lines;
+      intfun ();
+    with
+    | End_of_file -> close_in ich
+    | e -> close_in ich; raise e
+  in
+  intfun ();
+  List.rev !lines
+
+(*
+ * Same as lines_from_channel but from a file.
+ *)
+let lines_from_file fn =
+  let ich = open_in_bin fn in
+  let ls = lines_from_channel ich in
+  close_in ich; ls
+
+let chunk_reader_of_in_channel ich : chunk_reader =
+  function
+  | `Bytes num_bytes ->
+    begin try Ok (really_input_string ich num_bytes)
+      with
+      | End_of_file ->
+        let offset = pos_in ich in
+        close_in ich ;
```

Comment:
Do we really want to close the in_channel here? Shouldn't it be the responsibility of the creator of the chunk_reader to call close on it?

And if we do want to close the channel, we probably want the close_in_noerr here